flow/project.py

```diff
@@ -3519,8 +3469,20 @@ def _get_aggregate_from_id(self, id):
                 return aggregate_store[id]
             except KeyError:
                 pass
-        # Raise error as didn't find the id in any of the stored objects
-        raise LookupError(f"Did not find aggregate with id {repr(id)}.")
+        if check_abbrevations:
+            # No direct match was found, so check for abbreviated ids. Requires
+            # iteration over all elements of all aggregate stores.
+            matches = set()
+            for aggregate_store in self._group_to_aggregate_store.inverse:
+                for full_id in aggregate_store:
+                    if full_id.startswith(id):
+                        matches.add(aggregate_store[full_id])
```

Comment:
This implementation is linear in time. I wonder if we can make aggregation search using ids logarithmic by storing aggregate ids in a sorted manner in the aggregate stores? This is surely a tradeoff between time and space, hence, I think this should be a point to discuss in our next office hours.
PS: This is not completely related to this PR, so feel free to merge this one.

Reply:
The optimal (or at least the best standard) solution for this particular use-case would be to implement a trie for the aggregate stores, but that is definitely out of scope of this PR and we'd also need to make sure that it doesn't degrade performance in any other way. It's certainly an interesting idea though, and if you want to brainstorm at office hours that would be cool.